src/core/view/overlays/ShapeToolView.h

```diff
@@ -43,6 +43,14 @@ class ShapeToolView final: public BaseShapeOrSplineToolView, public xoj::util::L
     } FLAG_DIRTY_REGION = {};
     void on(FlagDirtyRegionRequest, const Range& rg);
 
+    /**
+     * @brief Called before the corresponding BaseShapeHandler's destruction
+     * @param rg The bounding box of the entire stroke
+     */
```

Comment:
Shouldn't the doc comment go on the function? Also, I think the `Finalization` struct itself also needs a doc comment.